File: ss-won/programmers/42628.cpp
#include <string>
#include <vector>
#include <deque>
#include <algorithm>
using namespace std;

vector<int> solution(vector<string> operations)
{
    vector<int> answer(2, 0);
    deque<int> dq;

    for (auto operation : operations)
    {
        char op = operation[0];
        int num = stoi(operation.substr(2));
        if (op == 'I')
        {
            dq.push_back(num);
            sort(dq.begin(), dq.end());
        }
        else
        {
            if (!dq.empty())
            {
                if (num > 0)
                    dq.pop_back();
                else
                    dq.pop_front();
                sort(dq.begin(), dq.end());
            }
        }
    }
    if (dq.empty())
        return answer;
    answer[0] = dq.back();
    answer[1] = dq.front();

    return answer;
}

File: ss-won/programmers/42628.py
def solution(operations):
    answer = [0, 0]
    q = []
    for operation in operations:
        op, num = operation.split()
        if op == 'I':
            q.append(int(num))
            q.sort()
        else:
            if len(q) == 0:
                continue
            q.sort()
            if int(num) > 0:
                q.pop(-1)
            else:
                q.pop(0)

    if len(q) == 0:
        return answer
    answer[0], answer[1] = q[-1], q[0]
    return answer
